src/ansys/aedt/core/emit_core/nodes/emit_node.py

```diff
@@ -551,6 +551,154 @@ def _is_column_data_table(self):
+                            row_list[i] = consts.unit_converter(val, "Frequency", unit, input_unit)
+                        elif unit != input_unit:
+                            raise ValueError(f"{unit} are not valid units for this property.")
+                        else:
+                            row_list[i] = val
+                    else:
+                        # Handle columns with SI units
+                        for unit_type, valid_units_list in EMIT_VALID_UNITS.items():
+                            if unit in valid_units_list:
+                                valid_unit = True
+                                unit_system = unit_type
+                                break
+                        if valid_unit:
+                            row_list[i] = self._convert_to_internal_units(value, unit_system)
+                        else:
+                            raise ValueError(f"{unit} are not valid units for this property.")
+                else:
+                    # Handle numeric inputs
+                    row_list[i] = value
+            data_return.append(tuple(row_list))
+        return data_return
+
+    def _check_node_prop_table_data(self, data):
+        """Converts user inputted int or string table data to SI units.
+
+        Parameters
+        ----------
+        data : list of tuples
+            User inputted table data.
+
+        Returns
+        -------
+        list of tuples
+            Data converted to SI units.
+        """
+        units = self._get_property("TableUnitTypes", True)
+        cols = self._get_property("TableColumns", True)
+
+        data_return = []
+        for row in data:
+            row_list = list(row)
+            if len(row) > len(units):
+                raise ValueError(f"Row {row} has more columns than expected ({len(units)}).")
+            for i, val in enumerate(row):
+                # Extract column unit if present in column header
+                col_unit = None
+                if "(" in cols[i]:
+                    col_unit = cols[i][cols[i].find("(") + 1 : cols[i].find(")")]
+
+                    # Update unit type based on column unit
+                    if col_unit in EMIT_VALID_UNITS["Frequency"]:
+                        units[i] = "FrequencyUnit"
+                    elif col_unit in EMIT_VALID_UNITS["Power"]:
+                        units[i] = "PowerUnit"
+
+                if "(" in cols[i] and isinstance(val, str):
+                    # Check for function input strings
+                    if "rf" in val.lower():
+                        row_list[i] = val
+                        continue
+
+                    # Process input values and units
+                    s = val.strip().replace(" ", "")
+                    unit_index = s.find(next(filter(str.isalpha, s)))
+                    value = float(s[:unit_index])
+                    input_unit = s[unit_index:]
+
+                    # Exception for dBc and dBm/Hz units
```

Comment:
```suggestion
                    # Handle dBc and dBm/Hz units
```

My fault for calling them "exceptions", "special cases" is probably better so people viewing the code don't expect "Exceptions" to be thrown when the cases are encountered.